app/Http/Controllers/MessageController.php

```diff
@@ -5,27 +5,43 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\DB;
 use App\Models\Message;
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Auth;
+use Symfony\Component\HttpFoundation\Response;
 
 class MessageController extends Controller
 {
     public function index(Request $request)
     {
-        $messages = DB::table('messages')->get();
-
-        return response()->json($messages);
+        $messages = Message::query()
+                ->leftjoin('users', 'user_uuid', '=', 'users.uuid')
```

Comment:
これめっちゃ初心者がハマりやすい沼なんですが，Eloquent Builder での（結果としての Eloquent Model の形を維持したままの） JOIN は，**サブクエリでの絞り込み用途以外では厳禁** です。 **「Message の下に User がぶら下がっている」という構造にならないからです。**

ではどうすればいいかというと，ここもリレーション機能に乗るのが正解です。

```suggestion
                ->with('user')
```

Reply:
元々レスポンスの構造を
```
{
  "status": 200,
  "data": [
  {
    "uuid": メッセージのuuid
    "name": 投稿者名
    "user_uuid": 投稿ユーザーのuuid
    "message": 本文
    "like_count": 0
    "like_status": bool
  },
  ]
}
```
のように設計してたんですが、
```
{
  "status": 200,
  "data": [
  {
    "uuid": メッセージのuuid
    "user_uuid": 投稿ユーザーのuuid
    "message": 本文
    "like_count": 0
    "like_status": bool
    "user": [
      {
      "name": 投稿者名
      ("uuid": 投稿ユーザーのuuid)
      }
    ]
  },
  ]
}
```
とした方が、messageにuserがぶら下がっている形としては自然なのでしょうか？

Reply:
あ，もとから設計がそうであったのであれば別に問題はないです！
意図せず使ってしまっていることを懸念してました

> messageにuserがぶら下がっている形としては自然なのでしょうか？

これは諸説あるのでおまかせしますｗ